### DISCUSSION TITLE
EBAI light curve fit failing

### DISCUSSION THREAD

**Discussion by TeHanHunter** [phoebe-project/phoebe2#479] [Questions]
Hello,

I am following the [inverse problem tutorial](inverse_paper_examples). I use interpolated TESS data and it worked for LC geometry estimator. 
![lc_geom](https://user-images.githubusercontent.com/49893001/124090603-90d12d80-da87-11eb-9293-b65361182806.jpg)
For EBAI, 

> b.add_solver('estimator.ebai', lc_datasets='lc01')
b.run_solver(kind='ebai', solution='ebai_sol')
print(b.adopt_solution('ebai_sol', trial_run=True))

It returns a Value Error:

> ValueError: failed to pass checks
Run Checks Report: FAIL
WARNING: solution='ebai_sol' is not included in run_checks_solution@setting, so will not raise interactive warnings (1 affected parameters, affecting )
ERROR: at least one parameter in adopt_parameters includes nan in fitted_values (2 affected parameters, affecting adopt_solution)

What is causing these nan values? Thank you!

**Comment by kecnry** [comment]
The [ebai estimator](http://phoebe-project.org/docs/latest/api/phoebe.parameters.solver.estimator.ebai) currently only supports well-detached systems (eclipse widths up to 0.25 in phase), although this may be extended in future releases as we get a better training set for the neural network.  So looking at your plot, I'm guessing that is what is happening in your case.  If you have a warning logger enabled it should include that in the message, but I'll also add a note to the tutorial you linked to avoid future confusion.

If only the inclination is returning nan (you can see this by printing the `fitted_values` parameter), then that is caused by the neural network returning a value for `sini` which fails when applying `np.arcsin`.  In this case, you can just exclude inclination from `adopt_parameters` and still adopt the rest of the proposed values.

**Reply by TeHanHunter** [reply]
Thank you! It seems that all fitted values from EBAI return nan. To continue the routine on the tutorial, I am trying to skip and work on the Nelder-Mead optimizer with

> b.add_compute('ellc', compute='fastcompute')
b.add_solver('optimizer.nelder_mead', fit_parameters=['teff@primary','requiv@primary', 'incl@binary', 'q', 'ecc', 'per0'],compute='fastcompute')
b.set_value('ld_mode@primary@lc01', 'lookup')
b.set_value('ld_mode@secondary@lc01', 'lookup')
b.run_solver(kind='nelder_mead', maxiter=10000, solution='nm_sol')

It runs very slowly and I wish to ask what parameters should I put in in this case (when I do not have teffratio, requivsumfrac, and inclination from EBAI).

**Reply by kecnry** [reply]
My general advice is to start with only a few parameters and include those that you think are most sensitive to the remaining residuals and then slowly add more or different parameters.  Its also useful to sometimes disable some higher-order expensive effects (like irradiation or roche distortion if its a well-detached system), at least during the initial optimization and then re-consider whether you need to include them as you progress.

This [tutorial from our recent workshop](http://phoebe-project.org/workshops/2021june/Tutorial_optimizing_computations) discusses some of these considerations in more detail.

**Reply by TeHanHunter** [reply]
I will look at that. Thank you!
